src/ui/Quickview/Quickview.ts

```diff
@@ -378,29 +378,49 @@ export class Quickview extends Component {
     };
   }
 
-  private prepareQuickviewContent(loadingAnimation: HTMLElement | Promise<HTMLElement>): Promise<Dom> {
-    return this.options.contentTemplate.instantiateToElement(this.result).then((built: HTMLElement) => {
-      const content = $$(built);
-
-      const initOptions = this.searchInterface.options;
-      const initParameters: IInitializationParameters = {
-        options: initOptions,
-        bindings: this.getBindings(),
-        result: this.result
-      };
-      return Initialization.automaticallyCreateComponentsInside(content.el, initParameters).initResult.then(() => {
-        if (content.find('.' + Component.computeCssClassName(QuickviewDocument)) != undefined && this.options.enableLoadingAnimation) {
-          if (loadingAnimation instanceof HTMLElement) {
-            content.prepend(loadingAnimation);
-          } else if (loadingAnimation instanceof Promise) {
-            loadingAnimation.then(anim => {
-              content.prepend(anim);
-            });
-          }
-        }
-        return content;
-      });
-    });
+  private async prepareQuickviewContent(loadingAnimation: HTMLElement | Promise<HTMLElement>): Promise<Dom> {
+    const domContent = await this.instantiateTemplateToDom();
+
+    const initOptions = this.searchInterface.options;
```

Comment:
What is a good general rule about temporary variables? 
Is it ok to use them in some scenarios or it is preferable to not use them?

Reply:
In that particular instance, it might not be needed indeed, it's mostly that it might not be clear that `searchInterface.options` actually refers to "an option that we need for initialization of components inside the quickview". 

So it's just a way to give it a "nice name" inside that function so that it's easier to read.